core/payments/vault/payment_vault.go

```diff
@@ -0,0 +1,91 @@
+package vault
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/Layr-Labs/eigenda/common"
+	bindings "github.com/Layr-Labs/eigenda/contracts/bindings/PaymentVault"
+	"github.com/Layr-Labs/eigenda/core/payments"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// Provides access to PaymentVault contract
+type paymentVault struct {
+	logger              logging.Logger
+	paymentVaultBinding *bindings.ContractPaymentVault
```

Comment:
recommend you switch to using abigen v2 if you have the time/energy. Ethen has made it easy to switch with his latest PR https://github.com/Layr-Labs/eigenda/pull/1910. You can check https://github.com/Layr-Labs/eigenda/pull/1911 for an example migration